src/backend/joanie/core/models/products.py

```diff
@@ -450,6 +478,59 @@ def is_enabled(self):
 
         return start <= now <= end
 
+    @property
+    def is_assignable(self):
+        """
+        Returns boolean whether the order group is enabled, and have available seats.
+        """
+        return self.is_enabled and self.available_seats != 0
+
+    def save(self, *args, **kwargs):
+        """
+        Override save method to assign the next available position number
+        within its course product relation if not already set.
+        """
+        if not self.created_on and self.position is None:
+            self.position = self.course_product_relation.order_groups.count()
```

Comment:
In case where a position is defined at creation. We should call `set_position` nope otherwise we could two order groups with the same position nope?

Reply:
Honestly it's an hazardous implementation, as I don't know how it should be managed on the frontend side.
Real test should give us directions on this.